packages/component/src/components/app-sidebar/app-updater-button/index.tsx

```diff
@@ -1,35 +1,111 @@
 import { useAFFiNEI18N } from '@affine/i18n/hooks';
-import { ResetIcon } from '@blocksuite/icons';
+import { CloseIcon, NewIcon, ResetIcon } from '@blocksuite/icons';
+import { assertExists } from '@blocksuite/store';
 import clsx from 'clsx';
+import { atom, useAtomValue, useSetAtom } from 'jotai';
+import { useEffect, useTransition } from 'react';
 
+import { changelogCheckedAtom, updateReadyAtom } from '../index.jotai';
 import * as styles from './index.css';
 
 interface AddPageButtonProps {
   className?: string;
   style?: React.CSSProperties;
 }
 
+const currentVersionAtom = atom(async () => {
+  const currentVersion = await window.apis?.updater.currentVersion();
+  return currentVersion;
+});
+
+const currentChangelogUnreadAtom = atom(async get => {
+  assertExists(window.appInfo);
+  const mapping = get(changelogCheckedAtom);
+  const currentVersion = await get(currentVersionAtom);
+  if (currentVersion) {
+    return !mapping[currentVersion];
+  }
+  return false;
+});
+
 // Although it is called an input, it is actually a button.
 export function AppUpdaterButton({ className, style }: AddPageButtonProps) {
   const t = useAFFiNEI18N();
 
+  useEffect(() => {
+    window.apis?.updater.checkForUpdatesAndNotify();
+  }, []);
+
+  const currentChangelogUnread = useAtomValue(currentChangelogUnreadAtom);
+  const updateReady = useAtomValue(updateReadyAtom);
+  const currentVersion = useAtomValue(currentVersionAtom);
+  const onReadOrDismissChangelog = useSetAtom(changelogCheckedAtom);
+
+  const [, transition] = useTransition();
+
+  const onReadOrDismissCurrentChangelog = (visit: boolean) => {
+    if (visit) {
+      window.open('https://github.com/toeverything/AFFiNE/releases', '_blank');
+    }
+
+    transition(() =>
+      onReadOrDismissChangelog(mapping => {
+        return {
+          ...mapping,
+          [currentVersion!]: true,
```

Comment:
Does `!` necessary here?

Reply:
Yeah the type inferred from `currentVersionAtom` is `string|undefined`. We could also use assertsExist to bypass it though.